lib/keycard/request/shibboleth_attributes.rb

```diff
@@ -12,32 +12,77 @@ module Request
     class ShibbolethAttributes < Attributes
       def base
         {
-          user_pid:  user_pid,
-          user_eid:  user_eid,
-          client_ip: client_ip
+          user_pid:     user_pid,
+          user_eid:     user_eid,
+          client_ip:    client_ip,
+          email:        email,
+          display_name: display_name,
+          affiliation:  affiliation,
+          authn_method: authn_context,
```

Comment:
This seems like an error, should be:
```ruby
authn_method: authn_method
authn_context: authn_context
```

Reply:
I thought about this a little... I didn’t think we would be operating in a SAML1 mode; that we pass the old attribute for backward compatibility. Correct?

For the naming, AuthnContextClassRef is truly awful, leaking gobs of Java and XML semantics.

Really, AuthenticationMethod was also pretty leaky, looking at why it was chosen, but not nearly as offensively abstract and long (even with two abbreviations). It sort of means what it says on the label, distinct from the implementation roots.

That is all in defense of choosing a convenient key that implies: this is how the IdP authenticated the user. I’m not even actually sure that an app should care, unless that’s how we check for step-up/MFA. There’s some value in indirection here, either way.

Maybe it drops out of base and just goes to verbatim?

All that said, I’m not vigorously opposed to including them both in base in direct form — just wanted to have some discussion.

Reply:
Agreed about authzContextClassRef / authzMethod - I don't know that we really need to be providing both, or really even either right now. At some future point I expect we'll make the determination in keycard whether 2FA was used, but since Fulcrum doesn't have that need right now (I think) I think I'd be happier not including those.